src/api/middleware/utils.js

```diff
@@ -0,0 +1,56 @@
+const HTTPStatus = require("http-status-codes");
+const { APIError, ErrorTypes } = require("./errorHandler");
+const ValidationReasons = require("./validators/validationReasons");
+const lodash = require("lodash");
+
+const DEFAULT_ERROR_CODE = ErrorTypes.VALIDATION_ERROR;
+const DEFAULT_ERROR_MSG = ValidationReasons.UNKNOWN;
+const DEFAULT_STATUS_CODE = HTTPStatus.BAD_REQUEST;
+
+/**
+ * Comibnes array of middleware using OR logic. Only fails if ALL functions fail (either by throwing or calling next(error))
+ *
+ * Each middleware will receive a different req object, no not rely on it to be shared among them
+ *
+ * @param {Function[]} Array of express middleware to be run
+ * @param {object} Options:
+ *  - error_code: error code in case of error (default: ErrorTypes.VALIDATION_ERROR)
+ *  - msg: the message in case of error (default: ValidationReasons.UNKNOWN)
+ *  - status_code: The status used in the HTTP Response in case of error (default: BAD_REQUEST (400))
+ */
+const or = (
+    [...middlewares],
+    {
+        error_code = DEFAULT_ERROR_CODE,
+        msg = DEFAULT_ERROR_MSG,
+        status_code = DEFAULT_STATUS_CODE
+    } = {}
+) => async (initialReq, res, next) => {
+    let success = false;
+    const errors = [];
+    for (const middleware of middlewares) {
+        const req = lodash.cloneDeep(initialReq);
+        if (success) return next();
+        try {
+            await middleware(req, res, (error) => {
+                if (error) errors.push(error);
+                else success = true;
```

Comment:
That would be great.... If that wasn't a lambda function :P

I had that at first, but returning there won't return on the outer function, so it will continue the execution. Having a success flag is necessary to pass the information to the outer function that it should stop executing

Reply:
Oh yeah, right! I totally missed that. Maybe after the try then?

Reply:
Yep, simplified a bit since we don't need the check on the last lines of the function

Reply:
Great